apps/condo/domains/banking/tasks/importBankTransactions.js

```diff
@@ -0,0 +1,198 @@
+
+        // User can cancel the task at any time, in this all operations should be stopped
+        task = await bankSyncTaskUtils.getOne(context, { id: taskId })
+        const taskStatus = get(task, 'status')
+        if (!task || taskStatus !== TASK_PROCESSING_STATUS) {
+            logger.info({ msg: 'status != processing. Aborting processing bank transactions loop', taskStatus, taskSchemaName: bankSyncTaskUtils.gql.SINGULAR_FORM, taskId })
+            return
+        }
+        
+        const transactionData = bankTransactionsData[i]
+        const existingTransaction = await BankTransaction.getOne(context, {
+            number: transactionData.number,
+            date: transactionData.date.toISOString(),
+        })
+        if (existingTransaction) continue
+        const payload = {
+            ...DV_SENDER,
+            number: transactionData.number,
+            date: transactionData.date.format('YYYY-MM-DD'),
+            isOutcome: transactionData.isOutcome,
+            purpose: transactionData.purpose,
+            currencyCode: 'RUB',
+            amount: transactionData.amount.toString(),
+            importId: transactionData.number,
+            importRemoteSystem: _1C_CLIENT_BANK_EXCHANGE,
+            organization: { connect: { id: organization.id } },
+            account: { connect: { id: bankAccount.id } },
+            integrationContext: { connect: { id: integrationContext.id } },
+            meta: transactionData.meta,
+        }
+        if (transactionData.contractorAccount) {
+            let existingContractorAccount = await BankContractorAccount.getOne(context, {
+                organization: { id: organization.id },
+                number: transactionData.contractorAccount.number,
+                tin: transactionData.contractorAccount.tin,
+            })
+            let contractorAccountId
+            if (existingContractorAccount) {
+                contractorAccountId = existingContractorAccount.id
+            } else {
+                const newContractorAccount = await BankContractorAccount.create(context, {
+                    ...DV_SENDER,
+                    ...transactionData.contractorAccount,
+                    country: organization.country,
+                    currencyCode: 'RUB',
+                    organization: { connect: { id: organization.id } },
+                })
+                contractorAccountId = newContractorAccount.id
+            }
+            payload.contractorAccount = { connect: { id: contractorAccountId } }
+        }
+
+        const createdBankTransaction = await BankTransaction.create(context, payload)
+        transactions.push(createdBankTransaction)
+
+        if (Date.now() - lastProgress > TASK_PROGRESS_UPDATE_INTERVAL) {
+            lastProgress = Date.now()
+            task = await bankSyncTaskUtils.update(taskId, {
+                ...DV_SENDER,
+                processedCount: i + 1,
+            })
+        }
+
+        await sleep(SLEEP_TIMEOUT)
```

Comment:
Chaining is complex to implement, because we need to store parsed "1CClientBankExchange" file somewhere to make each tasks be able to get theirs data and next data. Parsed data can be quite large and may not fit into Redis.